### ISSUE TITLE
Remove RH Question from Obstetric History and add logic from ANC Labs [6h]

### ISSUE THREAD

**Issue by adamhstewart** [TIP-Global-Health/eheza-app#1314]
## Issue

We should determine a patient's RH Status from the ANC Labs rather than the from the obstetric history.

![Pasted_Image_9_16_24__10_13_AM](https://github.com/user-attachments/assets/8af42800-3ec1-4451-a676-2964effdb228)

Instead we will capture the information we need from here in ANC Labs (Blood Group):
![Pasted_Image_9_16_24__10_16_AM](https://github.com/user-attachments/assets/368ab1e4-b81e-43fe-97ef-04b43fc77da8)

Results should be displayed in Obstetric Diagnosis section of the progress report with the following logic:
- If no test results show: RH Factor Unknown
- If test Result is negative, show: RH Factor Unknown
- If test result shows positive, show nothing
<img width="1125" alt="E-Heza_System_🔊" src="https://github.com/user-attachments/assets/97a6e7cb-1200-4c12-a779-eef33231b0e9">

## Tasks
- Remove question under obstetric history
- Add logic to show on progress report based upon the ANC Labs (Blood group)

**Comment by anvmn** [comment]
@LudmilaGoichman please test on Nutrition env

**Comment by LudmilaGoichman** [comment]
### Found Issue:
1. This `RH Factor Unknown` result isn’t exist. If test isn't done, the unknown result won't appear in the progress report.

> Results should be displayed in Obstetric Diagnosis section of the progress report with the following logic:
> - If no test results show: RH Factor Unknown

2. In the issue description in the report states "RH Factor Negative," while the application indicates "Rhesus Negative." Is it okay?

<img width="354" alt="image" src="https://github.com/user-attachments/assets/337cef77-0d3c-4fee-a8fa-86173d85057d">

### Checked and Approved:
1. RH Status is removed from the obstetric history.

<img width="161" alt="image" src="https://github.com/user-attachments/assets/290c62c0-dd8d-4f01-afde-30ea3b34fe7c">

2. RH Factor Negative:

- Point of care:

<img width="294" alt="image" src="https://github.com/user-attachments/assets/3570295f-f802-49a1-a103-05a0cf6fa802">|<img width="356" alt="image" src="https://github.com/user-attachments/assets/a329ae36-0f1b-489f-84a0-fd0f8db754c8">|<img width="323" alt="image" src="https://github.com/user-attachments/assets/043f09d0-92bf-45f5-8bff-e679cf1dd347">|<img width="354" alt="image" src="https://github.com/user-attachments/assets/7c43bfc1-531c-480c-8eed-ae5526b0d235">
---|---|---|---

- Lab:

<img width="356" alt="image" src="https://github.com/user-attachments/assets/0f6de3ea-8f19-46d9-8baa-7d8050867597">|<img width="356" alt="image" src="https://github.com/user-attachments/assets/c0c8342b-ca48-4fa6-8423-636cf791d16e">|<img width="325" alt="image" src="https://github.com/user-attachments/assets/03aedb26-d5f8-4a0f-9263-156d9c304fb5">
---|---|---

<img width="353" alt="image" src="https://github.com/user-attachments/assets/c003f714-0559-468f-a96b-71b123e6d536">|<img width="354" alt="image" src="https://github.com/user-attachments/assets/8d6ee3f1-58a0-4469-92c5-ec1d9d9c424e">|<img width="354" alt="image" src="https://github.com/user-attachments/assets/378508ef-a7be-4e3b-914c-9d0a62a32134">
---|---|---

<img width="354" alt="image" src="https://github.com/user-attachments/assets/8ec52657-c29d-4d38-9226-3278c8219faf">|<img width="354" alt="image" src="https://github.com/user-attachments/assets/87dd4693-3ad8-4065-8643-3f412f2a0949">|<img width="354" alt="image" src="https://github.com/user-attachments/assets/eea1efc6-ae11-4bc5-9d3e-405115f94ff9">
---|---|---

3. RH Factor Positive:

- Point of care:

<img width="312" alt="image" src="https://github.com/user-attachments/assets/e5ed49b5-87a2-49c8-8c01-ad5e63e483fe">|<img width="354" alt="image" src="https://github.com/user-attachments/assets/38717c7b-5e66-4290-8553-290c0f5f4866">
---|---

- Lab:

<img width="354" alt="image" src="https://github.com/user-attachments/assets/fcf9442e-26c2-4f4e-bf6a-eb7a6aabf8dc">|<img width="354" alt="image" src="https://github.com/user-attachments/assets/ce6050a5-37d4-42bf-8dfd-2b0769937025">|<img width="324" alt="image" src="https://github.com/user-attachments/assets/4171ada5-8466-4b4a-adc2-8988a2092535">
---|---|---

<img width="356" alt="image" src="https://github.com/user-attachments/assets/90e5c5e7-f367-4ed2-a8f1-a79fcddc1e10">|<img width="356" alt="image" src="https://github.com/user-attachments/assets/5aedd2db-6f20-4796-ae13-d74e017bfbc4">|<img width="356" alt="image" src="https://github.com/user-attachments/assets/4741c3ed-f0f9-4443-b425-927e63590b05">|<img width="356" alt="image" src="https://github.com/user-attachments/assets/fbca8963-7596-4117-8274-43d6623abf12">
---|---|---|---

**Comment by adamhstewart** [comment]
@anvmn please check first point

**Comment by anvmn** [comment]
@adamhstewart 

> @anvmn please check first point

'Obstetrical diagnosis' pane shows info of diagnoses that were determined in current encounter, and the action that was taken as a result of that determination. BTW, diagnosis is derived from ANC lab - no relation to obstetric history question.
So, this part is for current encounter only. For the context, Blood group and rhesus test is performed only once. Therefore, requirements which (concern info from previous encounters) can not be implemented for this section.

The changes I made are at Obstetrical alerts part which is shown only for CHW.

**Comment by adamhstewart** [comment]
@anvmn RH Factor Uknown should be a default Obsteric Diagnosis on any patient who has not had an RH lab result (for any reason).

**Comment by anvmn** [comment]
@LudmilaGoichman please retest on Nutrition env

**Comment by LudmilaGoichman** [comment]
@anvmn 

- Nutrition env
- Android tablet

### Questions/Found Issues:
1. The diagnosis appears twice in the CHW report.

<img width="351" alt="image" src="https://github.com/user-attachments/assets/9a89ffdc-0cb9-4a4f-92da-34daf9060d52">

2. The diagnosis is different. Is it okay?

- nurse:

<img width="350" alt="image" src="https://github.com/user-attachments/assets/3425c8ae-7de8-46bd-92d4-9fdfff03ea1e">|<img width="350" alt="image" src="https://github.com/user-attachments/assets/dcee83bc-5c07-4792-ac72-3197982ccfc4">|<img width="350" alt="image" src="https://github.com/user-attachments/assets/925b49aa-dc7a-4783-8132-48fc70ce9c33">
---|---|---

- CHW: 
<img width="351" alt="image" src="https://github.com/user-attachments/assets/ef75a40d-b820-41c3-83c2-f691ebb9d53d">

3. CHW. Should the report appear after clicking on encounter?

https://github.com/user-attachments/assets/cec761fe-3df5-4ec7-970f-25f78bc61c3a

4. Nurse. When there is the diagnosis, the report opens, but when there is no diagnosis, the report doesn't open. Is it okay? 

https://github.com/user-attachments/assets/7775065b-e279-428a-958f-5944268c7062

5. CHW. I can't click on this button.

<img width="352" alt="image" src="https://github.com/user-attachments/assets/53e69a99-51de-4578-960d-593803948853">

### Checked and Approved:
1. RH Factor Negative:
- Point of care:
   - Nurse
   
   
<img width="291" alt="image" src="https://github.com/user-attachments/assets/b8b9f48d-60b6-4d05-981b-24a383159742">|<img width="291" alt="image" src="https://github.com/user-attachments/assets/171121e2-5751-4a7c-ad23-559a17375f68">|<img width="355" alt="image" src="https://github.com/user-attachments/assets/ad755584-dad1-4be8-aea4-923d98694100">|<img width="355" alt="image" src="https://github.com/user-attachments/assets/b0a7ce92-96b1-4b89-abdb-a1a342e7e041">
---|---|---|---
   
   - CHW

https://github.com/user-attachments/assets/cec761fe-3df5-4ec7-970f-25f78bc61c3a

- Lab
   - Nurse 
   
<img width="355" alt="image" src="https://github.com/user-attachments/assets/49ee3dce-5af2-4f2d-8cd8-de99029fb359">|<img width="355" alt="image" src="https://github.com/user-attachments/assets/cd791c84-2e00-4e23-b800-9911a2aaf45e">|<img width="355" alt="image" src="https://github.com/user-attachments/assets/b2a941cf-b2c9-4fb8-a0de-a34fbe0bc171">
---|---|---

<img width="355" alt="image" src="https://github.com/user-attachments/assets/7d984539-ccb1-4a19-b7dc-e12d01c5f78e">|<img width="355" alt="image" src="https://github.com/user-attachments/assets/5a7dde74-36c7-4df6-810b-10b0b85d2023">
---|---|

   - CHW

<img width="355" alt="image" src="https://github.com/user-attachments/assets/a43df8d7-49e1-4e1f-a3c7-53dec043e5b3">

2. RH Factor Positive:
- Point of care:
   - Nurse
   
   
<img width="292" alt="image" src="https://github.com/user-attachments/assets/4f65cd03-ee17-44ed-b4c4-436eb917696f">| <img width="355" alt="image" src="https://github.com/user-attachments/assets/e94c5b36-8298-44a4-a0b5-9592fa634960">
---|---

   - CHW 
<img width="355" alt="image" src="https://github.com/user-attachments/assets/fe0158ba-27cb-441d-b082-b9b6fadb2c34">

- Lab
   - Nurse 
   
<img width="355" alt="image" src="https://github.com/user-attachments/assets/f771ec39-2ab4-4c5b-93ef-36241b80edb9">| <img width="355" alt="image" src="https://github.com/user-attachments/assets/d238ad8a-d622-498a-b870-2fab16d813ac">|<img width="355" alt="image" src="https://github.com/user-attachments/assets/af85dca2-4da6-4898-8314-5ccfaf24332c">
---|---|---

   - CHW

<img width="355" alt="image" src="https://github.com/user-attachments/assets/02b94f1c-9862-4bc6-8b62-ca944345a0ed">

**Comment by adamhstewart** [comment]
@anvmn 

> 1. The diagnosis appears twice in the CHW report.

Let's consolidate these report items.

> 2. The diagnosis is different. Is it okay?

Rhesus negative and RH Negative is the same thing - but (per above) in the progress report it should only be indicated once.

> 3. CHW. Should the report appear after clicking on encounter?

Yes, that is by design

> 4. Nurse. When there is the diagnosis, the report opens, but when there is no diagnosis, the report doesn't open. Is it okay?

It should open in every case.

> 5. CHW. I can't click on this button.

Please investigate

**Comment by anvmn** [comment]
@adamhstewart 


> > 1. The diagnosis appears twice in the CHW report.
> 
> Let's consolidate these report items.

Fixed at #1345

> > 2. The diagnosis is different. Is it okay?
> 
> Rhesus negative and RH Negative is the same thing - but (per above) in the progress report it should only be indicated once.

Fixed at #1345

> 
> > 4. Nurse. When there is the diagnosis, the report opens, but when there is no diagnosis, the report doesn't open. Is it okay?
> It should open in every case.


Per current implementation, it does not open in any case.
In given example, once there's a diagnosis, we get next steps, and once next steps are completed, we're forwarded to progress report.
If no diagnosis - no next steps - no such action.
If we want behaviour to change, lest address it in separate issue.

> 
> > 5. CHW. I can't click on this button.
> 
> Please investigate

So, first of all Yes is always clickable. The other button is active if there's at least one completed nurse encounter. Completed means that end encounter button was pressed (and end date is set).
Otherwise, we assume that first encounter is in progress (at prenatal, encounter may last more that single day).
